checkup.go

```diff
@@ -359,6 +359,12 @@ type Storage interface {
 	Store([]Result) error
 }
 
+// StorageReader can read results from the Storage.
+type StorageReader interface {
+	Fetch(string) ([]Result, error)
+	GetIndex() (map[string]int64, error)
```

Comment:
Sorry I didn't get to this sooner. We can keep it the same for now, see how that goes.